code/script.js

```diff
@@ -232,94 +235,175 @@ const start = () => {
+          alert(
+              `Yes, the person does have ${value} hair! Keep all people with ${value}`
+          )
+      } else {
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+          alert(
+              `No, the person doesnt have ${value} hair! Remove all people with ${value}`
+          )
+      }
+  } else if (category === 'eyes') {
+      if (keep) {
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+          alert(
+              `Yes, the person does have ${value} eyes! Keep all people with ${value}`
+          )
+      } else {
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+          alert(
+              `No, the person doesnt have ${value} eyes! Remove all people with ${value}`
+          )
+      }
   }
 
   // Determine what is the category
   // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  
+    // for hair and eyes :
+      // charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      // charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    // for accessories and other
+      // charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+      // charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+  
 
-  // Invoke a function to redraw the board with the remaining people.
+  // Invoke a function to redraw the board with the remaining people.¨
+  // console.log(category)
+  generateBoard(charactersInPlay)  
+  
 }
 
+  
+
+
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
+  let confirmVar = confirm("Are you sure?")
+  // console.log(personToConfirm +' test')
+  if (confirmVar == true) {
+    // console.log('something3')
+    checkMyGuess(personToConfirm)
+  } else {
+    stop
+  }
+}
+function reloadF(){
+  window.location.reload()
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
+  let txt = ''
+  // console.log('something2')
+  let winOrLoseTextElement = document.querySelector('#winOrLoseText');
+  let winOrloseWrapperElement = document.querySelector('.win-or-lose-wrapper')
```

Comment:
Interesting use of the querySelectors here. Was there any particular reason you decided to use querySelectors instead of `getElementByID` and `getElementsByClassName` as you did with the variables at the start? 

For the purpose of these variables, I went with `getElementByID`. They tend to have a [faster performance](https://gomakethings.com/javascript-selector-performance/). Not suuuper fast, but still :) 

 If you needed to escalate to [a more elaborate rule](https://careerkarma.com/blog/javascript-queryselector-vs-getelementbyid/) to select elements further down the road, then yeah, I'd go for querySelector.